src/main/java/spleetwaise/address/model/AddressBook.java

```diff
@@ -76,6 +77,17 @@ public Optional<Person> getPersonByPhone(Phone phone) {
         return Optional.of(filteredPersonList.get(0));
     }
 
+    public Optional<Person> getPersonByIndex(Index index) {
+        requireNonNull(index);
+
+        ObservableList<Person> PersonList = getPersonList();
```

Comment:
Is there a reason why `getFilteredPersonList ` isn't used instead? Just a quick glance, this is my main concern. Yet to launch the app to test the functionality.

Reply:
I just tested it and can confirm that the program misbehaves if we filter the persons list and attempt to add a transaction after that. See below:

1. Filter the persons list using `find Bernice`.
2. Running `addTxn 1 ...` adds a transaction with "Alex Yeoh" instead of "Bernice Yu".

<img width="1499" alt="image" src="https://github.com/user-attachments/assets/e3aeba9d-96a6-4654-a311-ccd55b104d13">

Reply:
@Dino-Nuggies recommend to remove transaction list getter and just use `getFilteredPersonList` as it returns the current state of the UI person list